modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md

```diff
@@ -1 +1,57 @@
+
+Different from the above two contact models, the `coulomb` model accounts for the friction along the interface caused by the relative motion between nodes. This is modeled using the Coulomb law, which computes a tangential traction that is proportional to the normal traction via a slip condition:
+\begin{equation*}
+{t}_{T} = -\mu |t_N| \frac{\boldsymbol{v}_T}{||\boldsymbol{v}_T||}\cdot\tau \quad \text{if} \quad ||{t}_T||> \mu |t_N|,
+\end{equation*}
+where $\mu$ is the friction coefficient.
+
+### Contact Formulations
+
+To enforce the contact constraints, three types of formulations are available. These are `kinematic`, `penalty`, and `mortar`.
+The `kinematic` and `penalty` formulations have been developed for some time and turn out to be robust for many mechanical contact problems. Recent efforts have been put in developing a mortar-based contact formulation. In this documentation, however,  we focus on `kinematic` and `penalty` formulations. For readers who are interested in mortar-based mechanical contact, please go to [mortar constraint system](Constraints/index.md) for more information.
+
+The overall workflow of imposing a mechanical constraint is as follows: 1) Determine whether penetration occurs. If so, constrain displacement; 2) In the normal direction, eliminate the penetration; 3) In the tangential direction, slip must satisfy friction model, no slit is allowed for the `glued` model, no constraint is applied for the `frictionless` model, Kuhn-Tucker conditions must be satisfied for the `coulomb` model. The mechanical contact enforcement steps for the `glued` and `frictionless` models are summarized below:
+
+For `penalty`:
+
+|  Glued  | Frictionless |
+| ---------- | ------------ |
+|  $\boldsymbol{f}_c = k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{f}_c = \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_p)$ |
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c$    | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i\boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+For `kinematic`:
+
+|  Glued  | Frictionless |
+| ---------- | ------------ |
+|  $\boldsymbol{f}_c = -\boldsymbol{r}_{s_{copy}}$     | $\boldsymbol{f}_c = -\boldsymbol{n}\boldsymbol{n}^{T} \boldsymbol{r}_{s_{copy}}$  |
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s+\boldsymbol{f}_c+k_p(\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c + \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+Note the kinematic method uses penalty parameter only to penalize error. Therefore, the converged solution has no error due to penalty compliance
```

Comment:
Note that the kinematic

Reply:
only to penalize error -> only to enforce the condition that the slave node has the desired position on the master face, so the errors are very small in the converged solution. Unlike in penalty formulations, penalty compliance is not introduced on the contact surface.